### PR TITLE
update models to include ip restrictions feat

### PR DESCRIPTION
**Pull request by tdaroly**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [ ] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [ ] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your pull request is merged into main branch, a new pull request will be created to update `src/index.json` automatically.  
The precondition is to put your code inside this repository and upgrade the version in the pull request but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
containerapp

**Comment by zhoxing-ms** [comment]
Do you need to release a new extension version for this PR? If so, please update the history notes in `HISTORY.rst` and the version defined in the `setup.py`

**Review by zhoxing-ms — APPROVED** [review]

File: src/containerapp/setup.py [hunk 1]
```diff
@@ -17,7 +17,7 @@
 # TODO: Confirm this is the right version number you want and it matches your
 # HISTORY.rst entry.
 
-VERSION = '0.3.20'
+VERSION = '0.3.22'
```
Comment (zhoxing-ms) [inline src/containerapp/setup.py:20]: It seems that version `0.3.21` has not been released yet? Can this PR be released together in version `0.3.21`?

File: src/containerapp/HISTORY.rst [hunk 2]
```diff
@@ -2,6 +2,10 @@
 
 Release History
 ===============
+0.3.22
+++++++
+* Fix the empty IP Restrictions object caused running `az containerapp update` command on Windows with a pre existing .yaml file
```
Comment (zhoxing-ms) [inline src/containerapp/HISTORY.rst:7]: You may consider placing this history note in version `0.3.21`

**Review by zhoxing-ms — APPROVED** [review]

**Pull request merged**